src/evolve/poller.go

```diff
@@ -0,0 +1,79 @@
+package evolve
+
+import (
+	"context"
+
+	"github.com/warp-contracts/syncer/src/utils/config"
+	"github.com/warp-contracts/syncer/src/utils/task"
+
+	"gorm.io/gorm"
+)
+
+// Periodically gets new evolved contract sources which are not yet in the db
+type Poller struct {
+	*task.Task
+
+	db      *gorm.DB
+
+	// Evolve to be sent out
+	Output chan *string
+}
+
+func NewPoller(config *config.Config) (self *Poller) {
+	self = new(Poller)
+ 
+	self.Output = make(chan *string, config.Evolve.ChannelBufferLength)
+
+	self.Task = task.NewTask(config, "poller").
+		WithRepeatedSubtaskFunc(config.Evolve.PollerInterval, self.handleNew).
+		WithOnAfterStop(func() {
+			close(self.Output)
+		})
+
+	return
+}
+
+func (self *Poller) WithDB(db *gorm.DB) *Poller {
+	self.db = db
+	return self
+}
+
+func (self *Poller) handleNew() (repeat bool, err error) {
+	self.Log.Info("Checking for new evolved sources...")
+	ctx, cancel := context.WithTimeout(self.Ctx, self.Config.Evolve.PollerTimeout)
+	defer cancel()
+
+	// Gets new evolved contract sources
+	var evolvedContractSources []string
+	err = self.db.WithContext(ctx).
+		Raw(`SELECT evolve
+		FROM interactions i
+		LEFT JOIN contracts_src cs
+		ON cs.src_tx_id = i.evolve
+		WHERE evolve IS NOT NULL AND cs.src_tx_id IS NULL;`).
```

Comment:
Add a LIMIT, use the return value to re-run query if len(evolvedContractSources)==LIMIT.